src/openforms/formio/api/serializers.py

```diff
@@ -25,6 +26,14 @@ class TemporaryFileUploadSerializer(serializers.Serializer):
         use_url=False,
         validators=[MimeTypeValidator(), NoVirusValidator()],
     )
+    submission = serializers.HyperlinkedRelatedField(
+        view_name="api:submission-detail",
+        lookup_field="uuid",
+        queryset=Submission.objects.all(),
```

Comment:
can we set this to `none` and dynamically based on the request set the queryset?

```suggestion
        queryset=Submission.objects.none(),
```

```py
def get_fields(self):
    fields = super().get_fields()
    session = self.context["request"].session
    fields["submission"].queryset = Submission.objects.filter(uuid__in=session.get(SUBMISSIONS_SESSION_KEY, []))
    return fields
```

That makes it safe by default and only explicitly sets the allow-list for valid requests.

Reply:
I initially went for this but results in the Spectacular API view to fail as it probably accesses the `fields` property (which call `get_fields`), but without any request available.

Reply:
`self.context["view"].swagger_fake_view`, see https://drf-spectacular.readthedocs.io/en/latest/faq.html#my-get-queryset-depends-on-some-attributes-not-available-at-schema-generation-time